test/jdk/javax/xml/jaxp/datatype/8033980/SerializationTest.java

```diff
@@ -25,160 +25,295 @@
+        //Serialize the given xml Duration
+        final ByteArrayOutputStream baos2 = new ByteArrayOutputStream();
+        final ObjectOutputStream oos2 = new ObjectOutputStream(baos2);
+        oos2.writeObject(duration);
+        oos2.flush();
+        oos2.close();
+        // Now get a Base64 string representation of the serialized bytes.
+        final String base64dur = Base64.getEncoder().encodeToString(baos2.toByteArray());
 
-    public void testReadCal() {
-        try {
-            for (String javaVersion : JDK) {
-                XMLGregorianCalendar d1 = (XMLGregorianCalendar) fromFile(
-                        javaVersion + FILENAME_CAL);
-                if (!d1.toString().equalsIgnoreCase(EXPECTED_CAL)) {
-                    fail("Java version: " + javaVersion
-                            + "\nExpected: " + EXPECTED_CAL
-                            + "\nActual: " + d1.toString());
-                } else {
-                    success("testReadCal: read " + javaVersion + " serialized file, passed.");
-                }
+        // Create the Data for JDK under test.
+        gregorianCalendarAndDurationSerData[0] = new GregorianCalendarAndDurationSerData() {
+            @Override
+            public byte[] getGregorianCalendarByteArray() {
+                return baos.toByteArray();
             }
-        } catch (ClassNotFoundException ex) {
-            fail("testReadCal: " + ex.getMessage());
-        } catch (IOException ex) {
-            fail("testReadCal: " + ex.getMessage());
-        }
-    }
 
-    public void testReadDuration() {
-        try {
-            for (String javaVersion : JDK) {
-                Duration d1 = (Duration) fromFile(
-                        javaVersion + FILENAME_DURATION);
-                if (!d1.toString().equalsIgnoreCase(EXPECTED_DURATION)) {
-                    fail("Java version: " + javaVersion
-                            + "\nExpected: " + EXPECTED_DURATION
-                            + "\nActual: " + d1.toString());
-                } else {
-                    success("testReadDuration: read " + javaVersion + " serialized file, passed.");
-                }
+            @Override
+            public String getGregorianCalendarBase64() {
+                return base64;
             }
-        } catch (ClassNotFoundException ex) {
-            fail("testReadDuration: " + ex.getMessage());
-        } catch (IOException ex) {
-            fail("testReadDuration: " + ex.getMessage());
-        }
+
+            @Override
+            public byte[] getDurationBytes() {
+                return baos2.toByteArray();
+            }
+
+            @Override
+            public String getDurationBase64() {
+                return base64dur;
+            }
+        };
     }
 
     /**
-     * Create test files
-     *
-     * @param javaVersion JDK version
+     *Provide data for JDK version and Gregorian Calendar serialized bytes
```

Comment:
I would suggest adding a comment for the parameters that are being created by the DataProvider